src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/ALoadoutSynchronizer.cs

```diff
@@ -248,6 +246,22 @@ await foreach (var entry in _hashCache.IndexFolderAsync(location))
                 Size = entry.Size,
                 LastModified = path.FileInfo.LastWriteTimeUtc
             };
+
+            // And mark them as executable if necessary.
+            // These below are constants, so on platforms like Windows, the execute bit setting
+            // code won't even get JIT-ted at all!
+            // Don't use `path.FileSystem.OS` because that's not constant, and this is a hot path.
+            if (!OperatingSystem.IsLinux() && !OperatingSystem.IsMacOS())
+                continue;
```

Comment:
The entire reason we have `IOperatingSystem` is so we don't use the `OperatingSystem` from .NET. If you are concerned about performance, just add a `isUnix` boolean before the loop:

```csharp
var isUnix = _os.IsUnix();

foreach (var (path, entry) in toExtract)
{
  // ...
  if (!isUnix) continue;
}
```

The JIT should be smart enough to deal with this for loop and realize that `isUnix` never changes.

Reply:
Kind of.

The difference between the code I wrote, and using `var isUnix = _os.IsUnix();` is that `OperatingSystem` is a true (JIT) constant.

When you make the proposed change, the JIT will (if it thinks it may be profitable), clone the entire loop, and simply make a branch based on `isUnix`.

i.e. the resulting code produced by the JIT would look something more like (after translating back to C#):

```csharp
var isUnix = _os.IsUnix();
if (isUnix)
{
    // Update the resulting items with the new file times
    foreach (var (path, entry) in toExtract)
    {
        resultingItems[entry.To] = new DiskStateEntry
        {
            Hash = entry.Hash,
            Size = entry.Size,
            LastModified = path.FileInfo.LastWriteTimeUtc
        };
        
        var ext = path.Extension.ToString();
        if (ext is not ("" or ".sh" or ".bin" or ".run" or ".py" or ".pl" or ".php" or ".rb" or ".out"
            or ".elf")) continue;

        // Note (Sewer): I don't think we'd ever need anything other than just 'user' execute, but you can never
        // be sure. Just in case, I'll throw in group and other to match 'chmod +x' behaviour.
        var currentMode = path.GetUnixFileMode();
        path.SetUnixFileMode(currentMode | UnixFileMode.UserExecute | UnixFileMode.GroupExecute | UnixFileMode.OtherExecute);
    }
}
else
{
    // Update the resulting items with the new file times
    foreach (var (path, entry) in toExtract)
    {
        resultingItems[entry.To] = new DiskStateEntry
        {
            Hash = entry.Hash,
            Size = entry.Size,
            LastModified = path.FileInfo.LastWriteTimeUtc
        };
    }
}
```

Even though `IsUnix` should be a true constant.
This increases JIT time and produces code that will never be executed, on a hot path. I'm not sure if that's quite desirable.

Reply:
>  This increases JIT time and produces code that will never be executed, on a hot path. I'm not sure if that's quite desirable.

A single if/else in a method doesn't matter. This isn't inside a loop.

Reply:
You're also forgetting about instruction cache and pre-fetching. If this was AOT'd (which we're not planning for anymore), this would also be all be dead code, inflating binary size.

That said, I'm not too obsessed about this. If it bothers me too much, I'll fix this issue at the root (i.e. replace `IOSInformation`).